tm2/pkg/db/pebbledb/pebbledb.go

```diff
@@ -0,0 +1,305 @@
+package pebbledb
+
+import (
+	"bytes"
+	goerrors "errors"
+	"path/filepath"
+	"slices"
+
+	"github.com/cockroachdb/pebble"
+
+	"github.com/gnolang/gno/tm2/pkg/db"
+	"github.com/gnolang/gno/tm2/pkg/db/internal"
+)
+
+func init() {
+	dbCreator := func(name string, dir string) (db.DB, error) {
+		return NewPebbleDB(name, dir)
+	}
+	db.InternalRegisterDBCreator(db.PebbleDBBackend, dbCreator, false)
+}
+
+var _ db.DB = (*PebbleDB)(nil)
+
+type PebbleDB struct {
+	db *pebble.DB
+}
+
+func NewPebbleDB(name string, dir string) (*PebbleDB, error) {
+	return NewPebbleDBWithOpts(name, dir, &pebble.Options{})
```

Comment:
Just curious, what would be potential options we'd want to enable for Pebble in the future?
I know the defaults are alright as they are